skimage/exposure/_adapthist.py

```diff
@@ -76,23 +76,30 @@ def equalize_adapthist(image, ntiles_x=8, ntiles_y=8, clip_limit=0.01,
     """
     image = img_as_uint(image)
     image = rescale_intensity(image, out_range=(0, NR_OF_GREY - 1))
-    out = _clahe(image, ntiles_x, ntiles_y, clip_limit * nbins, nbins)
-    image[:out.shape[0], :out.shape[1]] = out
+    if ntiles_x or ntiles_y:
+        ntiles_x = ntiles_x or 8
```

Comment:
Note: This will resolve to `8` if `ntiles_x=0` is passed. The acceptable range is [1, 16], so this _shouldn't_ be an issue. Still, the classic 

``` python 
if ntiles_x is None:
    ntiles_x = 8
```
remains best practice.